### PR TITLE
People pulling/grabbing a defib target now gets shocked harmfully when the defib goes off

### PR DESCRIPTION
**Pull request by silicons**
:cl:
rscadd: Nanotrasen would like to remind crewmembers and especially medical personnel to stand clear of cadeavers before applying a defibrillator shock. (You get shocked if you're pulling/grabbing someone being defibbed.)
tweak: defib shock/charge sounds upped from 50% to 75%.
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
some part of me thinks 30 burn is too high hmmm...

**Comment by coiax** [comment]
CLEAR!

**Comment by ExcessiveUseOfCobblestone** [comment]
Shocking!

File: code/game/objects/items/defib.dm [hunk 1]
```diff
@@ -452,6 +452,11 @@
 			playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 100, 1, -1)
 			playsound(loc, 'sound/weapons/egloves.ogg', 100, 1, -1)
 			H.emote("scream")
+			if(isliving(H.pulledby))		//CLEAR!
+				var/mob/living/M = H.pulledby
+				M.electrocute_act(45, src)
+				M.visible_message("<span class='danger'>[M] is electrocuted by their contact with [H]!</span>")
+				M.emote("scream")
```
Comment (RemieRichards) [inline code/game/objects/items/defib.dm]: screaming robots.
how exotic.

**Comment by davethwave** [comment]
Well considering that your using your hands to really pull someone or grab someone 30 does seem a bit high... how about 20 instead?

**Comment by imsxz** [comment]
Add a check to make sure that the one getting shocked doesn't have shock proof hands. Wouldn't make too much sense to be wearing insulated gloves and *still* be shocked.

**Comment by coiax** [comment]
@imsxz except you can pull someone even if you're missing hands...

**Comment by davethwave** [comment]
but you can't pull anything if your handcuffed...

**Comment by davethwave** [comment]
hmm... I see a fix or tweak PR coming due to this.

**Comment by Spaceinaba** [comment]
NOW I CAN YELL CLEAR AND IT WILL MEAN SOMETHING

**Comment by ExcessiveUseOfCobblestone** [comment]
> Well considering that your using your hands to really pull someone or grab someone 30 does seem a bit high... how about 20 instead?

Punishing mistakes are 11fun11

**Comment by PKPenguin321** [comment]
dont even deal shock damage just straight up give them a heart attack

**Comment by Repukan** [comment]
Cadavers or corpses.

**Comment by kevinz000** [comment]
who cares

File: code/game/objects/items/defib.dm [hunk 2]
```diff
@@ -500,11 +505,15 @@
...
+				playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 75, 1, -1)
 				total_brute	= H.getBruteLoss()
 				total_burn	= H.getFireLoss()
-
-				var/failed = null
+				if(isliving(H.pulledby))		//CLEAR!
+					var/mob/living/M = H.pulledby
+					if(M.electrocute_act(30, src))
+						M.visible_message("<span class='danger'>[M] is electrocuted by their contact with [H]!</span>")
+						M.emote("scream")
```
Comment (Cyberboss) [inline code/game/objects/items/defib.dm]: duplicate code, make it a proc

File: code/game/objects/items/defib.dm [hunk 3]
```diff
@@ -452,6 +452,11 @@
 			playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 100, 1, -1)
 			playsound(loc, 'sound/weapons/egloves.ogg', 100, 1, -1)
 			H.emote("scream")
+			if(isliving(H.pulledby))		//CLEAR!
+				var/mob/living/M = H.pulledby
+				if(M.electrocute_act(45, src))
+					M.visible_message("<span class='danger'>[M] is electrocuted by their contact with [H]!</span>")
+					M.emote("scream")
```
Comment (Cyberboss) [inline code/game/objects/items/defib.dm]: .

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/defib.dm [hunk 4]
```diff
@@ -479,7 +484,7 @@
 	update_icon()
 	if(do_after(user, 30, target = H)) //beginning to place the paddles on patient's chest to allow some time for people to move away to stop the process
 		user.visible_message("<span class='notice'>[user] places [src] on [H]'s chest.</span>", "<span class='warning'>You place [src] on [H]'s chest.</span>")
-		playsound(get_turf(src), 'sound/machines/defib_charge.ogg', 50, 0)
+		playsound(get_turf(src), 'sound/machines/defib_charge.ogg', 75, 0)
```
Comment (Cyberboss) [inline code/game/objects/items/defib.dm]: playsound(src, ...

File: code/game/objects/items/defib.dm [hunk 5]
```diff
@@ -500,11 +505,15 @@
 			if(H.stat == DEAD)
 				H.visible_message("<span class='warning'>[H]'s body convulses a bit.</span>")
 				playsound(get_turf(src), "bodyfall", 50, 1)
-				playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+				playsound(get_turf(src), 'sound/machines/defib_zap.ogg', 75, 1, -1)
```
Comment (Cyberboss) [inline code/game/objects/items/defib.dm]: playsound(src, ...

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/defib.dm [hunk 6]
```diff
@@ -500,11 +505,15 @@
...
 				total_brute	= H.getBruteLoss()
 				total_burn	= H.getFireLoss()
-
-				var/failed = null
+				if(isliving(H.pulledby))		//CLEAR!
+					var/mob/living/M = H.pulledby
+					if(M.electrocute_act(30, src))
+						M.visible_message("<span class='danger'>[M] is electrocuted by their contact with [H]!</span>")
+						M.emote("scream")
+				var/failed
```
Comment (Cyberboss) [inline code/game/objects/items/defib.dm]: Unused var

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/defib.dm (hunk 6 above)
Comment (kevinz000) [reply code/game/objects/items/defib.dm]: wait what

Comment (kevinz000) [reply code/game/objects/items/defib.dm]: this is used..?

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**